### ISSUE TITLE
<leftmouse> doesn't play nice with tabline-plugin barbar

### ISSUE THREAD

**Issue by astier** [dstein64/nvim-scrollview#16]
## Info

- NVIM v0.5.0-dev+1099-g4d5dbea4f
- scrollview tag v1.2.0
- [barbar](https://github.com/romgrk/barbar.nvim) commit 42d7112

## Problem

buffer-/tab-line plugins like barbar show the opened buffers in the tabline. It is also possible to click on the tabs to switch to the respective buffer inside the focused window. When there are at least two buffers and two windows open one can switch the buffer of any window by clicking on the respective buffer-tab. However, when using barbar and scrollview together it doesn't work as expected.

If two buffers and two split-windows are opened and the window on the bottom is focused, then trying to open the other buffer inside the bottom-window by clicking on the buffer-tab doesn't change the buffer inside the focused bottom-window. Instead the buffer inside the top-window changes.

## Minimal init.vim

```vim
set runtimepath+=~/.local/share/nvim/plugins/barbar.nvim
set runtimepath+=~/.local/share/nvim/plugins/nvim-scrollview
set mouse=a
```

## Reproduce

1. Open two files `nvim foo bar`
2. Insert newlines into `foo` until the scrollbar appears.
3. Split window into two windows `:sp<cr>`
4. Change focus to bottom window `:<c-w>j`
5. Click with `<leftmouse>` the `bar`-tab in the bufferline

It can be seen that the buffer changes from `foo` to `bar` in the unfocused top-window instead in the focused bottom-window. If this is repeated without loading `scrollview` inside the minimal init.vim or if the scrollview-mouse-bindings are disabled then it works as expected.

## Expected

The buffer should change in the focused window instead in the top-window.

**Comment by dstein64** [comment]
Thanks for reporting this @astier!

This is fixed as of eac810d716a1fd10da5c39e77c366f627ce9ab78 and 8339c656bb810fbba83b5b673bc4c6208c688078.

My intention was to have the current window change to the window where there was scrollbar dragging (in the case where a scrollbar was dragged in the non-current window). My original implementation was too broad in scope, always changing the current window to the window of the mouse event. eac810d716a1fd10da5c39e77c366f627ce9ab78 updated this to only update the window if there was a scrollbar drag. Upon further thought, I've modified the code again in 8339c656bb810fbba83b5b673bc4c6208c688078 so that the current window is not changed after scrollbar dragging. This is consistent with e.g., PyCharm, where the current window remains active even after dragging the scrollbar of some other window.

**Comment by astier** [comment]
Thank you. It works.
